packages/admin-panel/src/autocomplete/ReduxAutocomplete.jsx

```diff
@@ -37,12 +39,27 @@ const ReduxAutocompleteComponent = React.memo(
     placeholder,
     helperText,
   }) => {
+    const [hasSetInitialSelection, setHasSetInitialSelection] = useState(false);
+
     React.useEffect(() => {
       return () => {
         onClearState();
       };
     }, []);
```

Comment:
I presume this is intentional, but what makes this different from:
```tsx
React.useEffect(onClearState, []);
```
?

Reply:
If you return a function from `useEffect()` that function will get run whenever the component is removed from the DOM (it's called the cleanup function). So in this case it will clear the autocomplete when it's removed from the DOM. Whereas `useEffect(onClearState, [])` would clear the autocomplete when it is first rendered. Possibly that might result with the same behaviour in practice, but clearing in the cleanup function feels a bit more elegant.

We should probably have a comment explaining this though, I didn't know what it was doing when I first saw it either.